### ISSUE TITLE
Display-Sync=no when display-resample is in use

### ISSUE THREAD

**Issue by ShadyCrab** [plexinc/plex-media-player#119]
It seems that even though the log is reporting that the option is being set
2015-11-02 11:21:07 [ DEBUG ] PlayerComponent.cpp @ 396 - cplayer: Setting option 'video-sync' = "display-resample" (flags = 192) -> 0 

Plex isn't reporting that it's in use (Debug OSD, reports Display-Sync=no) . Is there something missing in the logs that would explain why display-sync is apparently not in use? This seems to happen regardless of whether I am using the default settings (or any combo of settings for that matter) or the following line I created:

vo-defaults=opengl-cb:scale=ewa_ginseng:scale-antiring=0.75:cscale=ewa_ginseng:dither-depth=auto:temporal-dither:dscale=mitchell:sigmoid-upscaling:fancy-downscaling:deband:interpolation:tscale=oversample:scaler-resizes-only:pbo

I really need Display-Sync to be in use, as Interpolation works far better with it enabled. Any idea on how to solve this?

**Comment by wm4** [comment]
This can depend on a bunch of things:
- is the display refresh rate known? (you can check in the debug overlay)
- the file could have a framerate that doesn't much well with the display refresh rate
- the file has weird timestamps, which makes the player unable to determine the actual framerate, which disables dsiplay-sync
- there was a desync of some sort, which disables display-sync

**Comment by TheFireKahuna** [comment]
Um, Display-Fps currently has "- (0)" in the debug overlay for all videos (and as expected, all videos have this issue). Any idea what could cause that? madVR + MPC-HC reports 59.989, whether its exclusive or windowed, or windowed fullscreen.

GTX 960, i5 4960, Win 10

**Comment by wm4** [comment]
It means our display manager code is messing up again.

>Learned from a log that :scaler-resizes-only is a bad idea, for some reason it was reporting the window size as the same resolution of the video, not the display. Could lead to some undesirable results.

No, this should be fine. The size the renderer gets here is the actual pixel size of whatever it renders to.

**Comment by tru** [comment]
Let's start to collect the display manager problems in a linked issue and think about how we can improve it. It seems to be a frequent source of errors.

**Comment by wm4** [comment]
A common problem is that PMP doesn't "find" the current display, and thus can't tell player the display refresh rate.

**Comment by TheFireKahuna** [comment]
Just want to update that I installed the 1.0.2 preview and unfortunately it still doesn't detect the refresh rate.

**Comment by TheFireKahuna** [comment]
1.0.3 latest preview, it displays my refresh rate as 59.940. Big step forward in some ways.

However, my actual refresh rate is 72hz in the Nvidia control panel (its a custom resolution). Same deal if I set my display to a native 50hz (non-custom). Still reporting 59.940. I know there isn't a refresh rate switch occurring, since when I alt tab its still showing 72hz. I also know that PMP runs in fullscreen windowed, and the alt-tab happens too fast compared to an actual refresh rate switch (like in the NVidia control panel).

In other words, display-sync is enabled, a refresh rate is being reported, but its not the actual refresh rate. Its the default refresh-rate of the screen, not the refresh-rate currently in use.

**Comment by wm4** [comment]
Which build number is that exactly? I know we're having problems with display mode switching and framerate determination on Windows.

**Comment by FireKahuna** [comment]
Slight bump :), but in the new 1.1.4.393 preview I still have this issue with custom resolutions. My refresh rate is 72hz (a custom resolution). The display-fps is 0.000. This is with or without display-sync enabled. With or without hardware decoding.

If I set it to a standard resolution (60hz, 50hz) it does correctly report exactly 60.000 and 50.000. Is this correct? Is it meant to detect the current refresh rate (like 60.003) or the expected refresh rate?

This is still an issue, since neither 50hz or 60hz can properly sync to 24hz content without frame drops.

Its with Angle 2.1 (GLES 3.0) on the same NVidia GTX 960 on the Win 10 anniversary update.

**Comment by wm4** [comment]
That's unexpected. Can you post another playback log?

**Comment by FireKahuna** [comment]
Here:
https://mega.nz/#!t5kGTaJI!tGrnUU9hkUBzmjYIIBLtQLIlWpWPMfUOeL_2gL6TKKE

I noticed that a 1920 x 1080 @72 mode is not listed, nor is my @75 mode. It's clearly not detecting custom resolutions.

**Comment by wm4** [comment]
Yes, that's the problem. Your display mode is not in the list, but our code expects it to, so it returns nothing instead. As a consequence we don't know the current refresh rate.

Internally, this is rather annoying, because the display mode is identified by an integer index. We could probably work it around by adding the current mode to our internal list if it's not represented by an existing entry.
